### PR TITLE
fix: bring package more inline with the WebRTC spec

### PR DESCRIPTION
**Pull request by ThaUnknown**
there were numerous issues which prevented this from behaving like a proper polyfill, these changes are ALL REQUIRED in order for it to work as a drop-in replacement for W3C's WebRTC

the changes are documented in each individual's commits description

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
         "typescript": "^4.6.3"
     },
     "dependencies": {
+        "node-domexception": "^1.0.0",
```
Comment (murat-dogan) [inline package.json:80]: I don't prefer any other dependency actually. But if we have to then I am ok with that.

File: polyfill/RTCDataChannel.js [hunk 2]
```diff
@@ -10,13 +10,12 @@ export default class _RTCDataChannel extends EventTarget {
...
     #ordered;
 
-    // needs to be called with .bind(this)
-    onbufferedamountlow = createEmptyFunction();
-    onclose = createEmptyFunction();
-    onclosing = createEmptyFunction();
-    onerror = createEmptyFunction();
-    onmessage = createEmptyFunction();
-    onopen = createEmptyFunction();
+    onbufferedamountlow;
```
Comment (murat-dogan) [inline polyfill/RTCDataChannel.js:13]: If we are calling like ` this.XXX?.(e);` then this is better

Comment (ThaUnknown) [reply polyfill/RTCDataChannel.js:13]: yes, according to spec, these can be null, so we shouldnt initiate the

File: package.json (hunk 1 above)
Comment (ThaUnknown) [reply package.json:80]: this simply forces exposing of node-domexception, so if we want to support node version 14-17  [domexception was added in node 17]

Comment (murat-dogan) [reply package.json:80]: What is the situation for v12? We need to support >=v12

Comment (ThaUnknown) [reply package.json:80]: eventtarget doesnt exist in node <14

Comment (paullouisageneau) [reply package.json:80]: Node 12 went end-of-life more than one year ago, support could be removed (even node 14 is not maintained anymore), actually removing node 12 would make it possible to clean up the legacy NAPI thread-safe callback mess.

**Pull request merged**